Resync `html/rendering` from WPT Upstream

Ahmad-S792 opened the pull request:
#### 3a46c8b831c8f3efbea062a3aa56781369c13bd3
<pre>
Resync `html/rendering` from WPT Upstream
<a href="https://bugs.webkit.org/show_bug.cgi?id=303524">https://bugs.webkit.org/show_bug.cgi?id=303524</a>

Reviewed by Brandon Stewart.

Upstream commit: <a href="https://github.com/web-platform-tests/wpt/commit/cb0485e0505240c7fa264b6d0e0f78165432674b">https://github.com/web-platform-tests/wpt/commit/cb0485e0505240c7fa264b6d0e0f78165432674b</a>

* LayoutTests/imported/w3c/resources/resource-files.json:
* LayoutTests/imported/w3c/web-platform-tests/html/rendering/non-replaced-elements/phrasing-content-0/WEB_FEATURES.yml: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/rendering/non-replaced-elements/phrasing-content-0/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/html/rendering/replaced-elements/embedded-content-rendering-rules/WEB_FEATURES.yml: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/rendering/replaced-elements/embedded-content-rendering-rules/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/html/rendering/replaced-elements/embedded-content/WEB_FEATURES.yml: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/rendering/replaced-elements/embedded-content/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/html/rendering/replaced-elements/images/img-represents-image-size-before-partially-available-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/rendering/replaced-elements/images/img-represents-image-size-before-partially-available.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/rendering/replaced-elements/images/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/html/rendering/replaced-elements/the-select-element/select-display-inline-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/rendering/replaced-elements/the-select-element/select-display-inline-ref.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/rendering/replaced-elements/the-select-element/select-display-inline.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/rendering/replaced-elements/the-select-element/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/html/rendering/the-details-element/w3c-import.log:

&gt; Platform Specific Expectations:
* LayoutTests/platform/ios/TestExpectations:
* LayoutTests/platform/mac-wk1/TestExpectations:

Canonical link: <a href="https://commits.webkit.org/304071@main">https://commits.webkit.org/304071@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/a2a600d92a7e6809c1ff2af6f8324ba388b46efa

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/134460 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/6924 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/45692 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/141993 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/86442 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/7ff999b8-3c83-4e04-bc93-3c7131e0ad58) 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/7523 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/6790 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/102775 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/70030 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/0335544c-3032-439a-928f-82ba3bc27ff3) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/137407 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/5254 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/120516 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/83566 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/dfd4b228-936a-4691-9407-fa6bcd4f84de) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/5114 "Passed tests") | [✅ 🧪 api-mac-debug](https://ews-build.webkit.org/#/builders/165/builds/2735 "Passed tests") | [  ~~🛠 wpe-cairo-libwebrtc~~](https://ews-build.webkit.org/#/builders/166/builds/2635 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/114324 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/38657 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/144684 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/6603 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/39241 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/111179 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/6680 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/5544 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/111449 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/28269 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/4950 "Passed tests") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/116793 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/60427 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/6656 "Built successfully") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/34992 "Passed tests") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/6478 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/6715 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/6590 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

stwrt reviewed (approved):
LGTM.

There is a test time out for `imported/w3c/web-platform-tests/html/rendering/replaced-elements/images/img-represents-image-size-before-partially-available.html` in Mac-WK1.

Ahmad-S792 committed:
Resync `html/rendering` from WPT Upstream
https://bugs.webkit.org/show_bug.cgi?id=303524

Reviewed by Brandon Stewart.

Upstream commit: https://github.com/web-platform-tests/wpt/commit/cb0485e0505240c7fa264b6d0e0f78165432674b

* LayoutTests/imported/w3c/resources/resource-files.json:
* LayoutTests/imported/w3c/web-platform-tests/html/rendering/non-replaced-elements/phrasing-content-0/WEB_FEATURES.yml: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/rendering/non-replaced-elements/phrasing-content-0/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/html/rendering/replaced-elements/embedded-content-rendering-rules/WEB_FEATURES.yml: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/rendering/replaced-elements/embedded-content-rendering-rules/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/html/rendering/replaced-elements/embedded-content/WEB_FEATURES.yml: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/rendering/replaced-elements/embedded-content/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/html/rendering/replaced-elements/images/img-represents-image-size-before-partially-available-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/rendering/replaced-elements/images/img-represents-image-size-before-partially-available.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/rendering/replaced-elements/images/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/html/rendering/replaced-elements/the-select-element/select-display-inline-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/rendering/replaced-elements/the-select-element/select-display-inline-ref.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/rendering/replaced-elements/the-select-element/select-display-inline.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/html/rendering/replaced-elements/the-select-element/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/html/rendering/the-details-element/w3c-import.log:

> Platform Specific Expectations:
* LayoutTests/platform/ios/TestExpectations:
* LayoutTests/platform/mac-wk1/TestExpectations:

Canonical link: https://commits.webkit.org/304071@main
